From 7b70de51df68659d3789481cf2446f8bc01e5387 Mon Sep 17 00:00:00 2001
From: Andy Green <andy@openmoko.com>
Date: Sun, 13 Apr 2008 07:25:48 +0100
Subject: [PATCH] introduce-bq27000-battery-driver.patch

This is a driver for the bq27000 found in the Highcell A5
battery, and the platform device stuff for it for GTA02.  It
is a Power Supply Class battery device.

The driver doesn't contain an HDQ engine but accepts pointers
from the platform data to the HDQ action routines; our
platform data plugs it into the FIQ HDQ engine stuff.

The Power Supply class exposes the battery down /sys so you
can find out battery status by doing the equivalent of this
bash command

for i in capacity charge_full current_now present status technology temp time_to_empty_now time_to_full_now type voltage_now ; do echo -n "$i    " ; cat /sys/devices/platform/bq27000-battery.0/power_supply/bat/$i ; done

Here is the kind of result you get from a battery discharging

capacity    0
charge_full    1215585
current_now    183375
present    1
status    Discharging
technology    Li-ion
temp    276
time_to_empty_now    0
time_to_full_now    3932100
type    Battery
voltage_now    2761000

Note that temp is in 1/10 degrees C, other values are in uV,
uA, uW.  The time_to_* reported are bogus, but that is what
the battery actually reports.

We can make more mappings to entries in power_supply class
but this is enough to get started with.

Signed-off-by: Andy Green <andy@openmoko.com>
---
 defconfig-2.6.24      |    1 +
 drivers/power/Kconfig |    7 -------
 2 files changed, 1 insertions(+), 7 deletions(-)

diff --git a/defconfig-2.6.24 b/defconfig-2.6.24
index 6d750a5..9467f4b 100644
--- a/defconfig-2.6.24
+++ b/defconfig-2.6.24
@@ -1826,3 +1826,4 @@ CONFIG_PLIST=y
 CONFIG_HAS_IOMEM=y
 CONFIG_HAS_DMA=y
 CONFIG_GTA02_HDQ=y
+CONFIG_BATTERY_BQ27000_HDQ=y
diff --git a/drivers/power/Kconfig b/drivers/power/Kconfig
index 5ace872..8c50ecb 100644
--- a/drivers/power/Kconfig
+++ b/drivers/power/Kconfig
@@ -64,10 +64,3 @@ config GTA02_HDQ
 
 endif # POWER_SUPPLY
 
-config GTA02_HDQ
-	tristate "Neo Freerunner HDQ"
-	depends on MACH_NEO1973_GTA02 && FIQ && S3C2440_C_FIQ
-	help
-	  Say Y to enable support for communicating with an HDQ battery
-	  on the Neo Freerunner.  You probably want to select
-	  at least BATTERY_BQ27000_HDQ as well
-- 
1.5.6.5

